src/main/java/org/zalando/riptide/Actions.java

```diff
@@ -48,10 +53,53 @@
                 response.getHeaders().getLocation();
     }
 
+    public static ThrowingFunction<ClientHttpResponse, URI, IOException> contentLocation() {
+        return response ->
+                Optional.ofNullable(response.getHeaders().getFirst(CONTENT_LOCATION))
+                .map(URI::create)
+                .orElse(null);
+    }
+
     public static <X extends Exception> EntityConsumer<X, X> propagate() {
         return entity -> {
             throw entity;
         };
     }
 
+    /**
+     * Normalizes the {@code Location} and {@code Content-Location} headers of any given response by resolving them
+     * against the given {@code uri}.
+     *
+     * @param uri the base uri to resolve against
+     * @return a function that normalizes responses
+     */
+    public static ThrowingFunction<ClientHttpResponse, ClientHttpResponse, IOException> normalize(final URI uri) {
+        return response -> {
+            final HttpHeaders headers = new HttpHeaders();
+            headers.putAll(response.getHeaders());
+
+            Optional.ofNullable(headers.getLocation())
```

Comment:
Would it make sense to first check whether we received a relative location or content-location? Maybe it's not needed, when the client uses this function it probably already knows that the server will use a relative uri.